app/models/collection_item.rb

```diff
@@ -253,18 +221,17 @@ def user_allowed_to_destroy?(user)
     user.is_author_of?(self.item) || self.collection.user_is_maintainer?(user)
   end
 
-  def approve_by_user ; self.user_approval_status = APPROVED ; end
-  def reject_by_user ; self.user_approval_status = REJECTED ; end
-  def approved_by_user? ; self.user_approval_status == APPROVED ; end
-  def rejected_by_user? ; self.user_approval_status == REJECTED ; end
+  def approve_by_user
```

Comment:
Can we replaced all uses of this with `approved_by_user!`? Same for `approved_by_collection!`.

Reply:
No, because `approve_by_user` & `approve_by_collection` are primarily used in `approve_automatically`, which runs `before_save` if the `CollectionItem`. So replacing them with `approved_by_user!` & `approved_by_collection!`, which save the record, results in an infinite loop.